src/main/histFileMod.F90

```diff
@@ -3088,124 +3088,124 @@ subroutine hfields_1dinfo(t, mode)
-               long_name='gridcell latitude', units='degrees_north', ncid=ncid)
+               long_name='gridcell latitude', units='degrees_north', fill_value=spval, ncid=ncid)
 
           call ncd_defvar(varname='grid1d_ixy', xtype=ncd_int, dim1name=nameg, &
-               long_name='2d longitude index of corresponding gridcell', ncid=ncid)
+               long_name='2d longitude index of corresponding gridcell', ifill_value=ispval, ncid=ncid)
 
           call ncd_defvar(varname='grid1d_jxy', xtype=ncd_int, dim1name=nameg, &
-               long_name='2d latitude index of corresponding gridcell', ncid=ncid)
+               long_name='2d latitude index of corresponding gridcell', ifill_value=ispval, ncid=ncid)
 
           ! Define landunit info
 
           call ncd_defvar(varname='land1d_lon', xtype=ncd_double, dim1name=namel, &
-               long_name='landunit longitude', units='degrees_east', ncid=ncid)
+               long_name='landunit longitude', units='degrees_east', fill_value=spval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_lat', xtype=ncd_double, dim1name=namel, &
-               long_name='landunit latitude', units='degrees_north', ncid=ncid)
+               long_name='landunit latitude', units='degrees_north', fill_value=spval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_ixy', xtype=ncd_int, dim1name=namel, &
-               long_name='2d longitude index of corresponding landunit', ncid=ncid)
+               long_name='2d longitude index of corresponding landunit', ifill_value=ispval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_jxy', xtype=ncd_int, dim1name=namel, &
-               long_name='2d latitude index of corresponding landunit', ncid=ncid)
+               long_name='2d latitude index of corresponding landunit', ifill_value=ispval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_gi', xtype=ncd_int, dim1name=namel, &
-               long_name='1d grid index of corresponding landunit', ncid=ncid)
+               long_name='1d grid index of corresponding landunit', ifill_value=ispval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_wtgcell', xtype=ncd_double, dim1name=namel, &
-               long_name='landunit weight relative to corresponding gridcell', ncid=ncid)
+               long_name='landunit weight relative to corresponding gridcell', fill_value=spval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_ityplunit', xtype=ncd_int, dim1name=namel, &
                long_name='landunit type (vegetated,urban,lake,wetland,glacier or glacier_mec)', &
-                  ncid=ncid)
+                  ifill_value=ispval, ncid=ncid)
 
           call ncd_defvar(varname='land1d_active', xtype=ncd_log, dim1name=namel, &
-               long_name='true => do computations on this landunit', ncid=ncid)
+               long_name='true => do computations on this landunit', ifill_value=0, ncid=ncid)
```

Comment:
Here and in two other places below: Is there a reason why you used a fill value of 0 rather than ispval? I don't think it will matter in practice for these variables, but I always find it confusing when I'm looking at a netcdf file where the fill value can be confused with a valid value for the given data.

- [ ] Unless there's a reason why this needs to be 0, we should change this to `ifill_value=ispval` here and in 2 lines below.

Reply:
The reason that I am adding fill values here is because the pio_write_darray calls in ncdio_pio.F90.in specifies fill values.
Since those fill values do not match the default fill values for netcdf you must set the _Fill_value attribute and that setting must match.   So see for example line 1744 of ncdio_pio.F90

Reply:
Okay, thanks for the explanation. And I think I just answered my other question above: since these are ncd_log, I understand that we need a 0 or 1 value... and I see that you're keeping consistent with what is done in the pio_set_darray call. Really I don't think we want fill values for some of these variables, but that's a change for a later date. So I'm fine with your changes - thanks.